### PR TITLE
TELCODOCS:582 - Create release notes for Assisted Installer release 2.3 and add API policy

### PR DESCRIPTION
**Pull request by MirzWeiss**
Version(s): Main, 4.10, 4.11

Issue: [TELCODOCS:582](https://issues.redhat.com/browse/TELCODOCS-582)

Link to docs preview: https://deploy-preview-45219--osdocs.netlify.app/openshift-enterprise/latest/installing/installing_on_prem_assisted/installing-on-prem-assisted.html

Additional information:
This PR also includes content for [TELCODOCS:592](https://issues.redhat.com/browse/TELCODOCS-592). This story required adding a sentence about the Assisted Installer API policy. Since this entire section is new, and the infrastructure for it is set up in this PR, I combined both stories into a single PR.

Acked by QE.

### CODE REVIEW

File: installing/installing_bare_metal_assisted/installing-bare-metal-assisted.adoc [hunk 1]
```diff
@@ -0,0 +1,18 @@
+:_content-type: ASSEMBLY
+[id="installing-bare-metal-assisted"]
+= Installing a cluster on bare metal using the Assisted Installer
+include::_attributes/common-attributes.adoc[]
+:context: installing-bare-metal-assisted
+
+toc::[]
+
+You can install {product-title} on bare metal using the {ai-full} (AI). The {ai-full} provides you with more options for customization than an installer-provisioned installation, while still running pre-flight validations and using smart defaults.
```
Comment (amolnar-rh) [inline installing/installing_bare_metal_assisted/installing-bare-metal-assisted.adoc:9]: If an abbreviation is introduced for the Assisted Installer, I'd use the expanded name of it in the next sentence.

File: modules/assisted-installer-release-notes.adoc [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+:_content-type: REFERENCE
+[id="assisted-installer-release-notes_{context}"]
+= {ai-full} {ai-version} release notes
+
+[id="ai-release-notes-about-this-release_{context}"]
+== About this release
+
+These release notes track the development of {ai-full} {ai-version}.
+
+This product was previously released as a Technology Preview product and is now generally available and enabled by default in {cluster-manager-first}.
```
Comment (amolnar-rh) [inline modules/assisted-installer-release-notes.adoc:14]:
```suggestion
This product was previously released as a Technology Preview product and is now generally available and enabled by default in the {cluster-manager-first}.
```

File: modules/assisted-installer-release-notes.adoc [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+[id="ai-release-notes-bug-fixes_{context}"]
+== Bug fixes
+
+*  Previously, users could define `localhost` as a valid host name for all of their hosts. As a result, host names were not unique, and {ai-full} could not install the cluster. With this release, users can not complete the cluster installation if any of the hosts are named `localhost`. An error appears and users must rename the hosts.
+//(link:https://issues.redhat.com/browse/MGMT-8088[MGMT-8088])
+
+* Previously, in the *OpenShift Web Console troubleshooting* window, the *Optional* field remained blank when undefined, instead of displaying an IP address. With this release, the *Optional* field was removed.
+//(link:https://issues.redhat.com/browse/MGMT-9283[MGMT-9283])
+
+* Previously, when installing a cluster on vSphere, the {ai-full} created machines and machineSets for every VM. With this release, {ai-full} no longer creates machines or machineSets for user-provisioned VMs.
```
Comment (amolnar-rh) [inline modules/assisted-installer-release-notes.adoc:25]: Please expand VM on the first instance.

File: modules/assisted-installer-release-notes.adoc [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+These release notes track the development of {ai-full} {ai-version}.
+
+This product was previously released as a Technology Preview product and is now generally available and enabled by default in {cluster-manager-first}.
+
+[id="ai-release-notes-bug-fixes_{context}"]
+== Bug fixes
+
+*  Previously, users could define `localhost` as a valid host name for all of their hosts. As a result, host names were not unique, and {ai-full} could not install the cluster. With this release, users can not complete the cluster installation if any of the hosts are named `localhost`. An error appears and users must rename the hosts.
+//(link:https://issues.redhat.com/browse/MGMT-8088[MGMT-8088])
```
Comment (amolnar-rh) [inline modules/assisted-installer-release-notes.adoc:20]: Just a reminder to uncomment the links when you're good to go.

**Review by amolnar-gh — COMMENTED** [review]
Great work! I only have small nits :)

File: modules/assisted-installer-release-notes.adoc [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+
+* Previously, in the *OpenShift Web Console troubleshooting* window, the *Optional* field remained blank when undefined, instead of displaying an IP address. With this release, the *Optional* field was removed.
+//(link:https://issues.redhat.com/browse/MGMT-9283[MGMT-9283])
+
+* Previously, when installing a cluster on vSphere, the {ai-full} created machines and machineSets for every VM. With this release, {ai-full} no longer creates machines or machineSets for user-provisioned VMs.
+//(link:https://issues.redhat.com/browse/MGMT-9559[MGMT-9559])
+
+* Previously, if operators failed to install during an installation with {ai-full}, users received an error message and were directed to reset the cluster installation. With this release, if operators fail to install, the cluster is automatically degraded.
```
[the remark this answers is not in the document]
Comment (MirzWeiss) [reply modules/assisted-installer-release-notes.adoc:28]: We're not linking to any of the bugs. Found out halfway through writing that the project isn't public, and they don't have parallel BZs for these. Not linking to issues was the ultimate PM decision, and in the future, they'll create bugzillas for all bugs.

File: _topic_maps/_topic_map.yml [hunk 6]
```diff
@@ -267,6 +267,12 @@ Topics:
     File: installing-bare-metal-network-customizations
   - Name: Installing a user-provisioned bare metal cluster on a restricted network
     File: installing-restricted-networks-bare-metal
+- Name: Installing on bare metal with Assisted Installer
```
Comment (avishayt) [inline _topic_maps/_topic_map.yml:270]: Do we purposefully want to limit the scope to bare metal?

Comment (MirzWeiss) [reply _topic_maps/_topic_map.yml:270]: Currently isn't that the only available option for AI? We can't let users think they can use AI to install on any platform when it's only for bare metal. Down the line if that changes, then that's a discussion to have, but the scope IS limited at the moment...

File: modules/assisted-installer-release-notes.adoc (hunk 3 above)
[the remark this answers is not in the document]
Comment (MirzWeiss) [reply modules/assisted-installer-release-notes.adoc:25]: Did some digging. This bug is connected to installing in a vSphere environment. I checked their docs, and it looks like "machines" don't require capitalization or backticks, but `MachineSets` requires both. WDYT?

[the remark this answers is not in the document]
Comment (MirzWeiss) [reply modules/assisted-installer-release-notes.adoc:25]: This is the updated bug: 

Previously, when installing a cluster on vSphere, the {ai-full} created machines and `MachineSet` objects for every VM. With this release, {ai-full} no longer creates machines or `MachineSet` objects for user-provisioned VMs.

File: _topic_maps/_topic_map.yml (hunk 6 above)
Comment (romfreiman) [reply _topic_maps/_topic_map.yml:270]: Can we restate it's on-prem vs bare metal?
because VMs on KVM / vsphere will work just fine and supported

**Review by avishayt — COMMENTED** [review]
One comment, otherwise looks great

File: modules/assisted-installer-release-notes.adoc [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+
+[id="ai-release-notes-about-this-release_{context}"]
+== About this release
+
+These release notes track the development of {ai-full} {ai-version}.
+
+This product was previously released as a Technology Preview product and is now generally available and enabled by default in the {cluster-manager-first}.
+
+[id="ai-release-notes-bug-fixes_{context}"]
+== Bug fixes
```
[the remark this answers is not in the document]
Comment (MirzWeiss) [reply modules/assisted-installer-release-notes.adoc:17]: @nalhadef that's so weird... No idea. It also shows up normally in my text editor...

**Pull request merged**